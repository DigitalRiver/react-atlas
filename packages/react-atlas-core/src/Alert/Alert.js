import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";

class Alert extends React.PureComponent {
  constructor(props) {
    super(props);

    // component state
    this.state = {
      "visible": true
    };
  }

  _closeAlert = () => {
    if (this.props.onDismiss) {
      this.props.onDismiss(event);
    }
    this.setState({
      "visible": false
    });
  };

  render() {
    let { hidden, children, dismissible, type, className, style } = this.props;
    const alertClasses = cx({
      "alert": true,
      "success": type === "success",
      "warning": type === "warning",
      "danger": type === "danger"
    });
    return (
      <React.Fragment>
        {this.state.visible && 
          <div
            hidden={hidden}
            className={cx(className)}
            style={style}
            styleName={alertClasses}
          >
            {dismissible && 
              <div
                onClick={this._closeAlert}
                type="button"
                styleName="close"
                data-dismiss="alert"
                aria-label="Close"
              >
                <span aria-hidden="true">Ã—</span>
              </div>
            }
            {children}
          </div>
        }
      </React.Fragment>
    );
  }
}

Alert.propTypes = {
  /**
   * Any HTML element or React Component.
   */
  "children": PropTypes.node.isRequired,

  /**
   * An object, array, or string of CSS classes to apply to Button.
   */
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /**
   * When true, the Alert can be dismissed.
   */
  "dismissible": PropTypes.bool,

  /**
   * A boolean to hide or show the alert component.
   */
  "hidden": PropTypes.bool,

  /**
   * Function that will be executed on dismiss.
   */
  "onDismiss": PropTypes.func,

  /**
   * Pass inline styles here.
   */
  "style": PropTypes.object,

  /**
   * Will set the Alert's style.  One of: 'info', 'success', 'warning', 'danger'
   */
  "type": PropTypes.string
};

Alert.defaultProps = {
  "type": "info"
};

export default Alert;
