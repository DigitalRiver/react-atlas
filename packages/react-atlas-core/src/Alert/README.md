Alert:
    <Alert>
        This is a default Alert!
    </Alert>

Alert types:

	<div>

		<Alert type="success">
	        <strong>Success!</strong> This alert box indicates a successful or positive action.
	    </Alert>
	    <br />

	    <Alert type="info">
	        <strong>Info!</strong> This alert box indicates a neutral informative change or action.
	    </Alert>
	    <br />

	    <Alert type="warning">
	        <strong>Warning!</strong> This alert box indicates a warning that might need attention.
	    </Alert>
	    <br />

	    <Alert type="danger">
	        <strong>Danger!</strong> This alert box indicates a dangerous or potentially negative action.
	    </Alert>
	</div>

Dismissible Alerts:

	<div>	
		<Alert type="success" dismissible>
	        <strong>Success!</strong> This alert box indicates a successful or positive action.
	    </Alert>
	    <br />

	    <Alert type="info" dismissible>
	        <strong>Info!</strong> This alert box indicates a neutral informative change or action.
	    </Alert>
	    <br />

	    <Alert type="warning" dismissible>
	        <strong>Warning!</strong> This alert box indicates a warning that might need attention.
	    </Alert>
	    <br />

	    <Alert type="danger" dismissible>
	        <strong>Danger!</strong> This alert box indicates a dangerous or potentially negative action.
	    </Alert>
	</div>

onDismiss callback:

	<Alert dismissible onDismiss={function(){alert('The alert has been dismissed.');}}>
        <strong>Success!</strong> This alert box will trigger a callback function when dismissed.
    </Alert>