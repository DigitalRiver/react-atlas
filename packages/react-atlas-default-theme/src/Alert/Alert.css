@import '../styles/variables.css';

.alert {
	composes: default-text from '../styles.css';
  	composes: default-font from '../styles.css';
    composes: primary-gray-border from '../styles.css';
    composes: rounded from '../styles.css';
    composes: pad-2 from '../styles.css';
    composes: bg-body from '../styles.css';
    border-left: 5px solid var(--info);
}

.success {
    border-left-color: var(--success);
}

.warning {
    border-left-color: var(--warning);
}

.danger {
    border-left-color: var(--error);
}

.close {
    composes: bold from '../styles.css';
    composes: bg-transparent from '../styles.css';
    composes: pad-0 from '../styles.css';
    composes: cursor-pointer from '../styles.css';
    composes: border-none from '../styles.css';
	float: right;
    font-size: 2.2rem;
    opacity: .2;
    filter: alpha(opacity=20);
    position: relative;
    top: -10px;
    right: -2px;
    -webkit-appearance:none;
}

.close:hover {
	opacity: 1;
    filter: alpha(opacity=100);
}