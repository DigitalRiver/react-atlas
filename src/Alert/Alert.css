@import '../styles/variables.css';

.alert {
	composes: default-text from '../styles.css';
  	composes: default-font from '../styles.css';
    composes: primary-gray-border from '../styles.css';
    composes: rounded from '../styles.css';
    composes: pad-2 from '../styles.css';
    composes: bg-body from '../styles.css';
    border-left: 5px solid var(--info);
    display: flex;
    white-space: pre-wrap;
    align-items: center;
    word-break: break-all;
}

.success {
    border-left-color: var(--success);
}

.warning {
    border-left-color: var(--warning);
}

.danger {
    border-left-color: var(--error);
}

.close {
    composes: bold from '../styles.css';
    composes: bg-transparent from '../styles.css';
    composes: pad-0 from '../styles.css';
    composes: cursor-pointer from '../styles.css';
    composes: border-none from '../styles.css';
    display: inline-block;
    font-size: 2.2rem;
    opacity: .2;
    filter: alpha(opacity=20);
    margin-left: auto;
    -webkit-appearance:none;
}

.close:hover {
	opacity: 1;
    filter: alpha(opacity=100);
}

.hidden {
    display: none;
}