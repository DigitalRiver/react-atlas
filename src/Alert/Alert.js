import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import styles from "./Alert.css";
import CSSModules from "react-css-modules";

export class Alert extends React.PureComponent {
  constructor(props) {
    super(props);

    // component state
    this.state = {
      "visible": true
    };
  }

  _closeAlert = () => {
    if (this.props.onDismiss) {
      this.props.onDismiss(event);
    }
    this.setState({
      "visible": false
    });
  };

  render() {
    let { hidden, children, dismissible, type, className, style } = this.props;
    const alertClasses = cx({
      "alert": true,
      "success": type === "success",
      "warning": type === "warning",
      "danger": type === "danger",
      hidden
    });
    return (
      <React.Fragment>
        {this.state.visible && 
          <div className={cx(className)} style={style} styleName={alertClasses}>
            {children}
            {dismissible && 
              <div
                onClick={this._closeAlert}
                type="button"
                styleName="close"
                data-dismiss="alert"
                aria-label="Close"
              >
                <span aria-hidden="true">Ã—</span>
              </div>
            }
          </div>
        }
      </React.Fragment>
    );
  }
}

Alert.propTypes = {
  /**
   * Any HTML element or React Component.
   */
  "children": PropTypes.node.isRequired,

  /**
   * An object, array, or string of CSS classes to apply to Button.
   */
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /**
   * When true, the Alert can be dismissed.
   */
  "dismissible": PropTypes.bool,

  /**
   * A boolean to hide or show the alert component.
   */
  "hidden": PropTypes.bool,

  /**
   * Function that will be executed on dismiss.
   */
  "onDismiss": PropTypes.func,

  /**
   * Pass inline styles here.
   */
  "style": PropTypes.object,

  /**
   * Will set the Alert's style.  One of: 'info', 'success', 'warning', 'danger'
   */
  "type": PropTypes.string
};

Alert.defaultProps = {
  "type": "info"
};

export default CSSModules(Alert, styles, { "allowMultiple": true });
