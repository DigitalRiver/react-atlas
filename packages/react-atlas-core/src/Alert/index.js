import { default as AlertCore } from "./Alert";
export { AlertCore };
