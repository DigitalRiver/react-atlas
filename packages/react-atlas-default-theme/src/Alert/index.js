import * as AlertStyle from "./Alert.css";
export { AlertStyle };
