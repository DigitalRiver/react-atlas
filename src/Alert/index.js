import Alert from "./Alert.js";
export { Alert };
