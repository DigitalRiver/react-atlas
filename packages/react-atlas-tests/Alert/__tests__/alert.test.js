import React from "react";
import { AlertCore } from "../../../react-atlas-core/src/Alert/index";
import renderer from "react-test-renderer";

describe("Test Alert component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <AlertCore type="success">
          <strong>Success!</strong> This alert box indicates a successful or
          positive action.
        </AlertCore>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
