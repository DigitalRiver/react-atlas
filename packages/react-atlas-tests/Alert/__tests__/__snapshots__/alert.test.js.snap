// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Alert component renders correctly 1`] = `
<div
  className=""
  hidden={undefined}
  style={undefined}
  styleName="alert success"
>
  <strong>
    Success!
  </strong>
   This alert box indicates a successful or positive action.
</div>
`;
